src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -390,6 +390,14 @@ CATCH_RETURN()
         std::vector<DWRITE_SHAPING_TEXT_PROPERTIES> textProps(textLength);
         std::vector<DWRITE_SHAPING_GLYPH_PROPERTIES> glyphProps(maxGlyphCount);
 
+        // Get the features to apply to the font
+        auto features = _fontRenderData->DefaultFontFeatures();
+        DWRITE_FONT_FEATURE* featureList = features.data();
+        DWRITE_TYPOGRAPHIC_FEATURES typographicFeatures = { &featureList[0], gsl::narrow<uint32_t>(features.size()) };
+        DWRITE_TYPOGRAPHIC_FEATURES const* typographicFeaturesPointer = &typographicFeatures;
+        const uint32_t fontFeatureLengths[1] = { textLength };

```

Comment:
You don't need to explicitly write `[1]`. `[]` is enough. The creation of the span is not necessary as you already have the pointer for this array.
You can also use `std::size(fontFeatureLengths)` to get the number of elements in an array like that.

Reply:
Fixed, thanks!

Reply:
Wait, we actually need the span because otherwise we get a `Expression 'fontFeatureLengths': No array to pointer decay` error when we pass `fontFeatureLengths` into `GetGlyphs`

Reply:
@PankajBhojwani Try writing `&fontFeatureLengths[0]`, or using `#pragma warning(suppress : 26485)`.

In fact you technically don't even need the array and you could just directly pass `&textLength`. I know the linter might be nagging, but this isn't a public API we're writing, where a `gsl::span` would be required for safety. Since this happens local within a function it's easy to manage the "unsafety" the linter believes exists here.

Reply:
if you want to do it the "acceptable" and "blessed" way that appeases the analyzer...

```
&til::at(fontFeatureLengths, 0)
```

Reply:
yes, this is dumb. no, this is not a drill

Reply:
using `til::at` doesn't compile, trying `gsl::at`